### PR TITLE
delete duplicate service_taxonomy

### PR DESCRIPTION
**Pull request by shakilhossain1**

### CODE REVIEW

File: sequelize/migrations/20250801124935-delete_duplicate_service_taxonomy_data.js [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    for (const { service_id, taxonomy_name } of serviceTaxonomyMappings) {
+      const serviceTaxonomy = await models.ServiceTaxonomy.findOne({
+        where: {
+          service_id,
+        },
+      });
+
+      if (serviceTaxonomy && serviceTaxonomy.taxonomy_id === healthTaxonomy.id) {
+        await destroyInstance('System',serviceTaxonomy);
+      }
```
Comment (jbeard4) [inline sequelize/migrations/20250801124935-delete_duplicate_service_taxonomy_data.js:251]: @shakilhossain1 The logic here is incorrect for a number of reasons.

First you are only handling the Health taxonomy. You want to map the other categories:
* Mental Health -> Health
* Employment -> Other Service
* Legal Service -> Other Service

Alternatively, you could make the logic so that we delete any records where the taxonomy **IS NOT** one of (Mental Health, Employment, Legal Service).

The next issue is that you are using `ServiceTaxonomy.findOne`. Instead of `findOne`, you should retrieve all of the records, and filter them to find records with old taxonomy. Otherwise findOne might retrieve the newer record with the new taxonomy, and we wouldn't clean up the old record.

Can you please try writing this again? 

Also, can you please ensure you test this locally? Please use the following SQL to test it to ensure there are no longer duplicate services:

```sql
select s.name, t.name from organizations o
inner join locations l on l.organization_id = o.id
inner join service_at_locations sal on sal.location_id = l.id
inner join services s on sal.service_id = s.id
inner join service_taxonomy st on st.service_id = s.id
inner join taxonomies t on t.id = st.taxonomy_id
where o.name = <organization name>
```

Thanks.

File: sequelize/migrations/20250801124935-delete_duplicate_service_taxonomy_data.js [hunk 2]
```diff
@@ -233,15 +233,21 @@ module.exports = {
...
+      const serviceTaxonomies = await models.ServiceTaxonomy.findAll({
         where: {
           service_id,
         },
       });
 
-      if (serviceTaxonomy && serviceTaxonomy.taxonomy_id === healthTaxonomy.id) {
-        await destroyInstance('System',serviceTaxonomy);
+      for (const serviceTaxonomy of serviceTaxonomies) {
+        if (serviceTaxonomy.taxonomy_id === healthTaxonomy.id || serviceTaxonomy.taxonomy_id === otherServicesTaxonomy.id) {
```
Comment (jbeard4) [inline sequelize/migrations/20250801124935-delete_duplicate_service_taxonomy_data.js:248]: @shakilhossain1 This looks better - thank you for revising. Can you confirm you were able to test this locally?

Comment (shakilhossain1) [reply sequelize/migrations/20250801124935-delete_duplicate_service_taxonomy_data.js:248]: Yes. I tested it localy.

**Comment by jbeard4** [comment]
This has now been run against test environment.
